src/utils/validations/validator/validator.js

```diff
@@ -0,0 +1,36 @@
+const isLegacy = (func) => { return typeof func === 'object'; };
+
+const validator = validationFunctions => (value, props) => (
+  new Promise(async (resolve, reject) => {
+    let results = [];
+
+    if (Array.isArray(validationFunctions)) {
+      results = validationFunctions.map((func) => {
+        if (isLegacy(func)) {
+          const valid = func.validate(value, props);
+          if (valid) {
+            return Promise.resolve();
+          }
+          return Promise.reject(func.message());
+        }
+        return func(value, props);
+      });
+    } else if (isLegacy(validationFunctions)) {
```

Comment:
You could transform `validationFunctions` into an array at the top of the function if needed, rather than need to introduce this second branch.